src/solver/variable/economy/max-mrg-utils.h

```diff
@@ -0,0 +1,51 @@
+#pragma once
+
+#include "../state.h"
+
+namespace Antares::Solver::Variable::Economy
+{
+
+struct MaxMRGinput
```

Comment:
The code previously contains in **max-mrg.cpp**	was refactored : 
- function computeMaxMRG(...) : contains the algorithm to compute max margin.
  - first argument is the max margin to compute (**out**)
  - second argument is a **struct** containing all required data to compute the max margin (**in**)
- `struct MaxMRGinput` : contains data required to compute the max margin
- **class MaxMrgDataFactory** : factory that builds data required to compute max margin.
  This class is abstract.
  It has got are 2 implementations : one that builds data required for classic max margin, and one that builds data required for CSR max margin